openthread: Separate OpenThread Kconfigs from L2 layer

ArekBalysNordic opened the pull request:
Moved OpenThread-related Kconfigs from L2 layer to modules/openthread. All of those configs were not strictly related
to the L2 layer, so they fits better to OpenThread module.

This operation allows using OpenThread Kconfigs even if L2 layer is disabled.

Enabling NET_L2_OPENTHREAD also configures those configs by selecting the OPENTHREAD kconfig, so there is no change regarding backwards compatibility. The only change is that the Kconfigs related to Thread were moved to the modules
space, so their location in the menuconfig also changed.

Once it is done, a choice for setting different L2 implementations seems to be redundant.

ArekBalysNordic committed:
openthread: Separate OpenThread Kconfigs from L2 layer

Moved OpenThread-related Kconfigs from L2 layer to
modules/openthread. All of those configs were not strictly related
to the L2 layer, so they fits better to OpenThread module.
This operation allows using OpenThread Kconfigs even if L2 layer is
disabled.

Enabling NET_L2_OPENTHREAD also configures those configs by
selecting the OPENTHREAD kconfig, so there is no change regarding
backwards compatibility. The only change is that the Kconfigs
related to Thread were moved to the modules space, so their
location in the menuconfig also changed.

Once it is done, a choice for setting different L2 implementations
seems to be redundant.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>

kkasperczyk-no commented on subsys/net/l2/openthread/Kconfig:
```diff
@@ -2,110 +2,23 @@
-	select REBOOT
-	select ENTROPY_GENERATOR
-
+	imply NET_L2_PHY_IEEE802154
```
Why do we want to change select to imply for `NET_L2_PHY_IEEE802154`?

ArekBalysNordic replied:
I thought we would like to have this config editable if someone has a custom IEEE802.15.4 PHY, but I re-checked it, and another config, not this one, sets the PHY driver. I'll leave `select` here.

ArekBalysNordic committed:
doc: releases: add release note for OpenThread update

Add an entry to the release notes and migration guide regarding
moving OpenThread-related Kconfig options from
subsys/net/l2/openthred/Kconfig to modules/openthread/Kconfig.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>
